app/components/badge.js

```diff
@@ -66,15 +81,15 @@ export default class Badge extends PureComponent {
         this.width = width;
         const height = Math.max(e.nativeEvent.layout.height, this.props.minHeight);
         const borderRadius = height / 2;
-        this.refs.badgeContainer.setNativeProps({
+        this.setNativeProps({
             style: {
                 width,
                 height,
                 borderRadius
             }
         });
         setTimeout(() => {
-            this.refs.badgeContainer.setNativeProps({
+            this.setNativeProps({
```

Comment:
Does this not cause this to loop? I feel like this should trigger `onLayout` again which would set another timeout